src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNodeToEnd(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementIndexForAdd(index);
+        if (size == 0) {
+            addFirstNode(value);
+        } else if (index == 0) {
+            addNodeToStart(value);
+        } else if (index == size) {
+            addNodeToEnd(value);
+        } else {
+            Node<T> currentNode = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        return node.element;
```

Comment:
Don't create redundant variables. The variable 'node' is redundant here as you can directly return 'findNodeByIndex(index).element'.